include/eve/detail/function/simd/riscv/combine.hpp

```diff
@@ -0,0 +1,63 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch.hpp>
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/module/core/regular/if_else.hpp>
+
+namespace eve::detail
+{
+template<typename T, typename N>
+EVE_FORCEINLINE auto
+combine(rvv_ const&, wide<T, N> l, wide<T, N> h) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  using that_t = wide<T, typename N::combined_type>;
+
+  constexpr size_t combined_vl = N::combined_type::value;
+  if constexpr( eve::has_aggregated_abi_v<that_t> )
+  {
+    that_t that;
+    that.storage().assign_parts(l, h);
+    return that;
+  }
+  else
+  {
+    auto           wider_l        = bit_cast(l, as<that_t> {});
+    auto           wider_h        = bit_cast(h, as<that_t> {});
+    constexpr auto shift_size     = N::value;
+    that_t         wider_h_placed = __riscv_vslideup(wider_h, wider_h, shift_size, combined_vl);
```

Comment:
can't you just slide the l,h in position, why ifs? or doesn't work

Reply:
In asm you can just move physical registers where you need (if lmul >=1), but unfortunately, in intrinsics interface there is no way to put l,h. And for cases when lmul <1 you will still use this code.

Reply:
fine for now but if I come up with a way to do this better, before we merge this, I will request a change.